alf/algorithms/agent.py

```diff
@@ -155,9 +175,20 @@ def rollout(self, time_step: ActionTimeStep, state: AgentState, mode):
         new_state = AgentState()
         info = AgentInfo()
         observation = self._encode(time_step)
+
+        if self._goal_generator is not None:
+            goal_step = self._goal_generator.predict(
```

Comment:
Cannot generate costs for on-policy algorithms if the goal generator is trainable

Reply:
Updated the interface to handle future on-policy learning as well as trainable goal_generator.